### DISCUSSION TITLE
Generate textured microstructure from Random microstructure

### DISCUSSION THREAD

**Discussion by makeabhishek** [mtex-toolbox/mtex#1105] [Ask Anything] [answered]
Hi all,
Is there a way in MTEX to change Euler angles of random microstructure to textured one by providing the preferred orientation to grains. 
Say example: A random HCP crystal system, and I want to enhance the texture in the 0001 direction along the c-axis.

suggestions and answers are much appreciated.
Thanks
Abhishek

**Comment by kilir** [comment, marked as answer]
Hi,
you mean you have Mtex grain objects? In that case you could sample from a model odf of any type and set those orientations as meanOrientation.

Cheers,
Rüdiger

**Reply by makeabhishek** [reply]
Thanks, Rüdiger,
If you remember my previous quest #1026. It is related to that. I have a .csv file with Euler angles and I want to enhance the texture in the pole figure.

Regards
Abhishek

**Reply by kilir** [reply]
I thought you were talking about something with a microstructure. If you simply want to add orientations with a specific property - e.g. being derived from a given ODF - you can still simply model an ODF (see the ODF documentation) and use `discreteSample()` to get a random sample from it.

Cheers,
Rüdiger

**Reply by makeabhishek** [reply]
My apologies Rüdiger,
Let me rephrase my question in more details.
Actually, I work on wave propagation and I have not much knowledge of materials. My objective is to generate different texture material (modify Euler angles) by changing the texture (increase/decrease texture MUD) with similar morphology. I have Euler angles data set and I want to modify it to change the texture values (Pole figure shown below). 

For example, I have a given pole figure (plotted from the Euler angles data set) where the maximum intensity is around 15, as shown in the below pole figure. Now I want to change the texture (ex. intensity maybe 20, 10,5, etc ), I think if the intensity value will decrease it means having a low texture and vice versa.
Finally, after modifications, I want to generate new Euler angle data set, which I can use in my study.

So, how can I modify these texture values by any means? I'm not sure if this can be done by ODF or with some other tool. Kindly advice. 

![image](https://user-images.githubusercontent.com/47937684/123230655-0d08c580-d50a-11eb-8571-ebfca2fc991a.png)

Please correct me if I am wrong and provide your suggestion.

Thanks, 

Regards
Abhishek

**Reply by kilir** [reply]
I'd say that is exactly what I suggested above.
1) you model an odf with whatever shape/strength and property you desire (see `unimodalODF`, `fibreODF` or `BinghamODF`)
2) you sample as many orientations as you want from that odf (see `discreteSample`)
Hope that helps.
```
cs = crystalSymmetry('622')
odf = fibreODF(Miller(0,0,0,1,cs),xvector,'halfwidth',20*degree)
o = discreteSample(odf,1000)
```

Cheers,
Rüdiger

**Reply by makeabhishek** [reply]
Thanks, Rüdiger, 
I think this will work for me. May I ask a few fundamental questions, please?

1. What are the meaning of `halfwidth` and angle `20*degree` in `fibreODF()`. How these affect the results? I tried fullwidth also, and it works too. I used this code.

```
              cs = crystalSymmetry('6/mmm')
              odf = fibreODF(Miller(0,0,0,1,cs),zvector,'halfwidth',20*degree)
              o = discreteSample(odf,1000)
```

2. Curious, what's the symmetry '622' that you used in the above example.

3. Also, is it possible to plot the microstructure to see the grain morphology after changing the texture (Just like the EBSD image, maybe IPF maps), from the available data? How can I do that?


Thanks 
Abhishek

**Reply by makeabhishek** [reply]
@kilir I tried to follow the steps, but I realised that descreteSample is generating random points for a given texture component. I want a similar morphology compared with the original microstructure. 
For example, if there are elongated grains in the original microstructure,  it should also be in the modified texture.

Thanks
Regards
Abhishek

**Reply by kilir** [reply]
Hi,

there is no microstructure in a list of Euler angles or in an ODF.

Please make a larger effort to explain what you want.

Rüdiger
